src/realm/query.hpp

```diff
@@ -370,6 +372,7 @@ class Query final {
     LnkLstPtr m_source_link_list;                  // link lists are owned by the query.
     ConstTableView* m_source_table_view = nullptr; // table views are not refcounted, and not owned by the query.
     std::unique_ptr<ConstTableView> m_owned_source_table_view; // <--- except when indicated here
+    std::shared_ptr<DescriptorOrdering> m_ordering;
```

Comment:
It makes a lot of sense to me to attach the descriptors to the query itself. But there is now two ways to store this information, the other being in the TableView. Maybe it's worth adding a couple asserts in TableView if people try to apply a descriptor there and the view has been generated by a query?

Reply:
The descriptors are actually still stored in the TableView. I don't think we need to restrict what the user can to with the TableView after it has been created. He can add to them or remove them.